tests/MongoDB.Driver.Tests/MongoCollectionImplTests.cs

```diff
@@ -3429,6 +3429,7 @@ public void Watch_should_support_full_document_with_duplicate_elements()
             var client = DriverTestConfiguration.Client;
             var database = client.GetDatabase(DriverTestConfiguration.DatabaseNamespace.DatabaseName);
             var collection = database.GetCollection<BsonDocument>(DriverTestConfiguration.CollectionNamespace.CollectionName);
+            collection.InsertOne(new BsonDocument("_id", 1)); // ensure database exists
```

Comment:
On sharded systems only the database must exist before collection.Watch can be called.

Depending on the order the tests were run in the database might or might not already have existed, so this test was sometimes failing

Reply:
interesting

Reply:
The new race condition is that some previous test left a collection behind that already had a document with `{ _id : 1 }` in it...

Reply:
Minor note, it looks like it's enough to have only:
    
    new BsonDocument()

Reply:
That's true. That would just result in the `_id` field being generated automatically, and it would be an `ObjectId`.

I've generally leaned in the direction of avoiding hidden side effects.